packages/cardano-services/src/ChainHistory/DbSyncChainHistory/mappers.ts

```diff
@@ -91,29 +91,36 @@ export const mapTxInModel = (txInModel: TxInputModel): TxInput => ({
   txSourceId: txInModel.tx_source_id.toString('hex') as unknown as Cardano.TransactionId
 });
 
-export const mapTxOut = (txOut: TxOutput): Cardano.TxOut => ({
-  address: txOut.address,
-  datum: txOut.datum,
-  datumHash: txOut.datumHash,
-  scriptReference: txOut.scriptReference,
-  value: txOut.value
-});
+export const mapTxOut = (txOut: TxOutput) => {
+  const result: Cardano.TxOut = { address: txOut.address, value: txOut.value };
+
+  if (txOut.datum) result.datum = txOut.datum;
+  if (txOut.datumHash) result.datumHash = txOut.datumHash;
+  if (txOut.scriptReference) result.scriptReference = txOut.scriptReference;
+
+  return result;
```

Comment:
I'd prefer the suggested form as well, but I had to chose for the used approach to make the output exactly equal to the `DbSyncChainSyncProvider` ones.

In case of an output without any additional data, the `DbSyncChainSyncProvider` output is:
```
{
  "address": "abc",
  "value", 10n
}
```
while the output of the _preferred_ form would be:
```
{
  "address": "abc",
  "value", 10n,
  "datum": undefined,
  "datumHash": undefined,
  "scriptReference": undefined
}
```
I know there is no difference between the two outputs, but the comparison tests would need a lot of additional work to inspect each value rather just comparing the serialized form of the two outputs.

Reply:
Ill leave this to the team to decide, not my code, not my call

Reply:
The target here is to produce an output exactly equal to the `DbSyncChainSyncProvider` one to be sure the change has no impacts.
Refactoring the original `Provider` is out of the scope of this PR.

Reply:
I would leave data with `undefined` and do filtering to ignore undefined, probably there already is some assert for that. but its up to you. Feel free to resolve this